### PR TITLE
[los] Port Eden's line-of-sight system

### PR DESCRIPTION
**Pull request by zach2good**
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my PR will be ignored.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have _**tested my code and the things my code has changed**_ since the last commit in the PR and will test after any later commits.

## What does this pull request do?

- Add forked submodule of Eden's LOS meshes
- Port Eden's LOS code
- Add our-style CMake stuff for Eden's code
- Make the replacements in the code for raycasting and canSee, etc.
- Tidy some header includes while I'm in there
- run_clang_format.sh
- Add some startup logging for "Hey, I notice your navmesh or los folders are empty, just letting you know"
- Also brings in some commits with misc improvements I saw on Eden while I was poking around
- Profile (its fast! but it boosts full load time by 23s):
- image
- Test and verify locally!
- Find ways to speed up startup when single-process and loading 299 zones. It's now considerably slower.
  - Yes! parallel for_each loop saves the day!
image
- Mark zones with the new misc flags, if needed
- Eden uses this:
```sql
-- Make players unable to cast through walls in cities (1), dungeon zones (3), battlefields (4), instances (6), and limbus areas (7).
UPDATE zone_settings SET misc = misc + 0x1000 WHERE misc & 0x1000 = 0 AND zonetype IN (1, 3, 4, 6, 7);
-- Also disallow casting through walls in Dynamis cities and nyzul
UPDATE zone_settings SET misc = misc + 0x1000 WHERE misc & 0x1000 = 0 AND zoneid IN (42, 77, 185, 186, 187, 188, 294, 295, 296, 297);
```
## Steps to test these changes

Go get chased around by mobs, get drawn-in, cast spells etc.

### CODE REVIEW

**Comment by zach2good** [comment]
This PR is the proper implementation of https://github.com/LandSandBoat/server/pull/1474

and probably invalidates: https://github.com/LandSandBoat/server/pull/1483

**Comment by zach2good** [comment]
Would probably also be good to make sure that NMs will only draw-in to points that they can see, if that was still a problem?

File: tools/dbtool.py [hunk 1]
```diff
@@ -979,14 +979,23 @@ def configure_and_launch_multi_process_by_zonetype():
...
+
 def tasks_menu():
     present_menu(
         "Maintenance Tasks",
         {
-            "1": ["Set zone IP addresses", set_external_ip_dialog],
-            "2": ["Server-wide announcement", announce_menu],
-            "3": ["Show table sizes (min 2MB)", print_db_tables_by_size],
-            # "4": [
+            "1": ["Update git submodules", update_submodules],
```
Comment (claywar) [inline tools/dbtool.py:994]: Love having this baked into dbtool as well.  For future, we should probably update the wiki to instruct people to use it on first install and updates (even if we're not updating the submodules often)

File: src/map/ai/controllers/mob_controller.cpp [hunk 2]
```diff
@@ -1149,18 +1141,30 @@ bool CMobController::Cast(uint16 targid, SpellID spellid)
 bool CMobController::CanMoveForward(float currentDistance)
 {
     TracyZoneScoped;
-    if (PMob->m_Behaviour & BEHAVIOUR_STANDBACK && currentDistance < 20)
+
+    uint16 standbackRange = 20;
```
Comment (claywar) [inline src/map/ai/controllers/mob_controller.cpp:1145]: Out of scope for this PR, but does this ever change?

**Review by claywar — APPROVED** [review]

File: src/map/ai/controllers/mob_controller.cpp (hunk 2 above)
Comment (zach2good) [reply src/map/ai/controllers/mob_controller.cpp:1145]: I set it up this way to mimic what Eden is doing in this block; they have a mobmod for this and we dont, so I left it hard-coded

Comment (TeoTwawki) [reply src/map/ai/controllers/mob_controller.cpp:1145]: we probably should follow up with adding that mobmod - I am fairly sure there are a few mobs on retail that disobey the standard rules and that's also a fun customization point too. 

Note: we could also convert the one that handles if they standback at all, and  if we don't want them to, just set it zero, but that both creates a barrier to them eventually getting back in sync with us and is potentially a surprise bug situation later in existing code.

**Pull request merged**